skbio/stats/composition.py

```diff
@@ -2058,3 +2061,407 @@ def dirmult_ttest(
+    ...     index=["x1", "x2", "x3", "y1", "y2", "y3",
+    ...            "z1", "z2", "z3", "u1", "u2", "u3"],
+    ... )
+    >>> res = dirmult_lme(
+    ...     formula="time + treatment", data=table, metadata=metadata,
+    ...     groups="patient", seed=0
+    ... )
+    >>> res
+      FeatureID  Covariate  Log2(FC)   CI(2.5)  CI(97.5)    pvalue    qvalue
+    0        Y1       time -0.210769 -0.731153  0.309650  0.403737  0.403737
+    1        Y1  treatment -0.744093 -1.780094  0.291999  0.252057  0.252057
+    2        Y2       time  0.210769 -0.309650  0.731153  0.403737  0.403737
+    3        Y2  treatment  0.744093 -0.291999  1.780094  0.252057  0.252057
+
+    """
+
+    if not hasattr(data, "__getitem__"):
+        raise ValueError(
+            """data must be a pandas DataFrame or a numpy structured
+              or rec array or a dictionary"""
+        )
+
+    rng = get_rng(seed)
+
+    posterior = [
+        rng.dirichlet(data.values[i] + pseudocount) for i in range(data.shape[0])
+    ]
+
+    dir_table = pd.DataFrame(clr(posterior), index=data.index, columns=data.columns)
+
+    _call_res = _lme_call(
+        formula=formula,
+        table=dir_table,
+        metadata=metadata,
+        groups=groups,
+        reml=reml,
+        method=method,
+        fit_kwargs=fit_kwargs,
+        **kwargs,
+    )
+
+    res = _call_res[0]
+    list_of_vars = _call_res[2]
+
+    submodels = _call_res[1]
+    for i in submodels:
+        i = i.fit(reml=reml, method=method, **fit_kwargs)
+
+    for single_covar_data in res:
+        # multiple comparison
+        if p_adjust is not None:
+            qval = _calc_p_adjust(p_adjust, single_covar_data["pvalue"])
+        else:
+            qval = single_covar_data["pvalue"].values
+
+        single_covar_data["qvalue"] = qval[0]
+
+    # Creating an empty dict to store sum of values (Using a DataFrame threw errors)
+    # uses a separate array for each covariate and each feature
+    # array index: 0: pvalue, 1: CI(2.5), 2: CI(97.5), 3: qvalue, 4: log2fc
+    res_dict = {}
+
+    for b in data.columns:
+        res_dict[b] = {}
+        for covar in list_of_vars:
+            res_dict[b][covar] = [0, 0, 0, 0, 0]
+
+    for single_covar_data in res:
+        res_dict[single_covar_data["FeatureID"]][single_covar_data["Covariate"]][0] = (
+            single_covar_data["pvalue"]
+        )
+        res_dict[single_covar_data["FeatureID"]][single_covar_data["Covariate"]][1] = (
+            single_covar_data["CI(2.5)"]
+        )
+        res_dict[single_covar_data["FeatureID"]][single_covar_data["Covariate"]][2] = (
+            single_covar_data["CI(97.5)"]
+        )
+        res_dict[single_covar_data["FeatureID"]][single_covar_data["Covariate"]][3] = (
+            single_covar_data["qvalue"]
+        )
+        res_dict[single_covar_data["FeatureID"]][single_covar_data["Covariate"]][4] = (
+            single_covar_data["Log2(FC)"]
+        )
+
+    for i in range(1, draws):
```

Comment:
Something is weird with this ordering. Within each of the bootstraps, the model should be refitted.
Note the call to welch ttest : https://github.com/scikit-bio/scikit-bio/blob/main/skbio/stats/composition.py#L2011-L2016